parcel-bundler/parcel
# Use cargo workspaces

Use cargo workspaces [0] to manage our native Rust crates. This should result in less duplication of dependencies, leading to less downloading and more build cache overlap as we expand the number of crates in the repo. It also allows rust-analyzer [1] to easily detect the main Rust workspace.

This also updates npm's `@napi-rs/cli` to version 1.0.4 and cargo's `napi-rs` and `napi-derive` to their latest versions.

Test Plan: 
* Run `node ./scripts/build-native.js` and verify that `.node` files continue to be placed in the correct package directory so publishing continues to work.
* Verify that language features from rust-analyzer activate with the rust-analyzer VSCode extension, whereas they did not before.

[0] https://doc.rust-lang.org/cargo/reference/workspaces.html
[1] https://rust-analyzer.github.io/
